### PR TITLE
Update typescript to 5.3.3

### PR DESCRIPTION
**Pull request by znarf**
May fix https://github.com/depcheck/depcheck/issues/814

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,7 +101,7 @@
     "proxyquire": "^2.1.3",
     "should": "^13.2.3",
     "svelte": "^3.38.2",
-    "typescript": "^4.0.5"
+    "typescript": "^5.1.6"
```
Comment (vivi90) [inline package.json:104]: Upgrade to TypeScript 5 is not neccessary and may break some type definitions.
Please use `^4.9.0`.

**Review by vivi90 — CHANGES_REQUESTED** [review]
Please change to `^4.9.0` or `^4.9.5` instead.
But better to `^4.9.0` since that is the required version.

**Comment by rumpl** [comment]
> But better to `^4.9.0` since that is the required version.

Required for what?

>  TypeScript 5 may break some type definitions.

Can you be more specific please?

**Comment by znarf** [comment]
I think the tool should generally try to update its dependencies continuously to their latest version.

Here, if there is no test broken, we can assume that there is no breaking change and can move forward.

If we want to document a particular behavior and prevent against breaking changes, we should ideally have that covered with a test.

**Review by rumpl — APPROVED** [review]

**Pull request merged**